### PR TITLE
refactor(babel-preset-app):  use `@babel/helper-compilation-targets` instead of deprecated api

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Per https://github.com/babel/babel/blob/v7.8.7/packages/babel-preset-env/src/index.js#L33 and https://github.com/babel/babel/blob/v7.8.7/packages/babel-preset-env/src/targets-parser.js#L1, some apis are gonna be removed, we can use the new ones now.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/babel-preset-app/src/index.js [hunk 1]
```diff
@@ -32,15 +32,20 @@ function getDefaultPolyfills (corejs) {
 }
 
 function getPolyfills (targets, includes, { ignoreBrowserslistConfig, configPath, corejs }) {
-  const { isPluginRequired } = require('@babel/preset-env')
+  const { default: getTargets, isRequired } = require('@babel/helper-compilation-targets')
```
Comment (pi0) [inline packages/babel-preset-app/src/index.js]: Should we add direct `@babel/helper-compilation-targets` dependency?

Comment (clarkdo) [reply packages/babel-preset-app/src/index.js]: I think it's find, `@babel/helper-compilation-targets` is also used by `@babel/preset-env` and other packages

Comment (pi0) [reply packages/babel-preset-app/src/index.js]: As we directly require this module I think we need to define in package.json. At least it is a requirement of yarn v2 and similar new PMs.

Comment (clarkdo) [reply packages/babel-preset-app/src/index.js]: I'll add it.

Comment (clarkdo) [reply packages/babel-preset-app/src/index.js]: @pi0 Thanks

**Pull request merged**